BlizzHUDTweaks/MouseoverFrameFading.lua

```diff
@@ -207,22 +207,49 @@ end
 
 local function determineMouseOver(profile, frameName, frameOptions)
   local linkedFrames = profile[frameName .. "LinkedFrames"]
-  local linkedFrameMouseover
+  local mainFrame = frameOptions.mainFrame
+  local fallback = mainFrame.__BlizzHUDTweaksForceMouseover or mainFrame:IsMouseOver()
+
+  -- Check if SpellFlyout is shown and mouse-over
+  local spellFlyoutButtonBarName = ""
+  if SpellFlyout and SpellFlyout:IsShown() and SpellFlyout:IsMouseOver() then
+    local button = SpellFlyout:GetParent()
+    if button and button.bar then
+      spellFlyoutButtonBarName = button.bar:GetName()
+    end
+  end
+
+  local linkedFrameMouseover = false
 
+  -- Frame with links
   if linkedFrames then
     for linkedFrameName, _ in pairs(linkedFrames) do
-      if profile[linkedFrameName].Enabled then
-        if linkedFrames[linkedFrameName] and addon:GetFrameMapping()[linkedFrameName].mainFrame then
-          if addon:GetFrameMapping()[linkedFrameName].mainFrame:IsMouseOver() then
-            linkedFrameMouseover = true
-            return linkedFrameMouseover or frameOptions.mainFrame:IsMouseOver()
+      local frameProfile = profile[linkedFrameName]
+      if frameProfile and frameProfile.Enabled then
+        if linkedFrames[linkedFrameName] then
+          local mappedFrame = addon:GetFrameMapping()[linkedFrameName]
+          if mappedFrame and mappedFrame.mainFrame then
+            if spellFlyoutButtonBarName == mappedFrame.mainFrame:GetName() then
+              return true
```

Comment:
If we determine that a linked frame is being hovered over, we can return it immediately without having to possibly loop through all the other frames as well. Don't we?

Reply:
This comment was supposed to be on the line mentioned in the next comment, sorry for the confusion :D